iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/optional.inl

```diff
@@ -298,14 +286,56 @@ inline void optional<T>::destruct_value() noexcept
     value().~T();
     m_hasValue = false;
 }
-
+// AXIVION Next Construct AutosarC++19_03-M17.0.3 : make_optional is defined within iox::cxx and
+// is not overridden
 template <typename OptionalBaseType, typename... Targs>
 inline optional<OptionalBaseType> make_optional(Targs&&... args) noexcept
 {
-    optional<OptionalBaseType> returnValue = nullopt_t();
+    optional<OptionalBaseType> returnValue{nullopt_t()};
     returnValue.emplace(std::forward<Targs>(args)...);
     return returnValue;
 }
+
+template <typename T>
+bool operator==(const optional<T>& lhs, const optional<T>& rhs) noexcept
+{
+    auto bothNull = !lhs.has_value() && !rhs.has_value();
+    auto bothEqual = (lhs.has_value() && rhs.has_value()) && (*lhs == *rhs);
+    return bothNull || bothEqual;
+}
+
+template <typename T>
+bool operator!=(const optional<T>& lhs, const optional<T>& rhs) noexcept
+{
+    return !(lhs == rhs);
+}
+
+// AXIVION DISABLE STYLE AutosarC++19_03-A13.5.5: Comparison with nullopt_t for easier unset
```

Comment:
```suggestion
// AXIVION DISABLE STYLE AutosarC++19_03-A13.5.5: Comparison with nullopt_t is required
```

This is also a questionable rule. If one needs comparison with additional types as here, it should be allowed without having to write a comment... 

@FerdinandSpitzschnueffler @saif-at-github 

Probably OK here to disable it for the whole (short) section. I would favor to disable it globally, but we should not delay the PR with discussion about this.

Reply:
This should be fine for now. Thanks.